packages/core/src/utils.ts

```diff
@@ -223,9 +223,35 @@ export function objectEntries(object: { [key: string]: unknown }) {
   return Object.keys(object).map((key) => [key, object[key]])
 }
 
+/**
+ * inspired by https://mathiasbynens.be/notes/globalthis
+ */
 export function getGlobalObject<T>(): T {
-  // tslint:disable-next-line: function-constructor no-function-constructor-with-string-args
-  return (typeof globalThis === 'object' ? globalThis : Function('return this')()) as T
+  if (typeof globalThis === 'object') {
+    return (globalThis as unknown) as T
+  }
+  Object.defineProperty(Object.prototype, '_dd_temp_', {
+    get() {
+      return this
+    },
+    configurable: true,
+  })
+  // @ts-ignore
+  let globalObject: unknown = _dd_temp_
+  // @ts-ignore
+  delete Object.prototype._dd_temp_
+  if (typeof globalObject !== 'object') {
+    // on safari _dd_temp_ is available on window but not globally
+    // fallback on other browser globals check
```

Comment:
I've preferred to add this fallback instead of only checking all the possible globals, since the above method should be more robust.